airbyte-ci/connectors/pipelines/pipelines/dagger/actions/gradle.py

```diff
@@ -0,0 +1,230 @@
+            .from_(AMAZONCORRETTO_IMAGE)
+            # Mount the dependency cache volume, but not to $GRADLE_HOME, because gradle doesn't expect concurrent modifications.
+            .with_mounted_cache(self.GRADLE_DEP_CACHE_PATH, self.dependency_cache_volume, sharing=dagger.CacheSharingMode.LOCKED)
+            # Set GRADLE_HOME to the directory which will be rsync-ed with the gradle cache volume.
+            .with_env_variable("GRADLE_HOME", self.GRADLE_HOME_PATH)
+            # Same for GRADLE_USER_HOME.
+            .with_env_variable("GRADLE_USER_HOME", self.GRADLE_HOME_PATH)
+            # Install a bunch of packages as early as possible.
+            .with_exec(
+                sh_dash_c(
+                    [
+                        # Update first, but in the same .with_exec step as the package installation.
+                        # Otherwise, we risk caching stale package URLs.
+                        "yum update -y",
+                        f"yum install -y {' '.join(self.YUM_PACKAGES_TO_INSTALL)}",
+                        # Remove any dangly bits.
+                        "yum clean all",
+                        # Deliberately soft-remove docker, so that the `docker` CLI is unavailable by default.
+                        # This is a defensive choice to enforce the expectation that, as a general rule, gradle tasks do not rely on docker.
+                        "yum remove -y --noautoremove docker",  # remove docker package but not its dependencies
+                        "yum install -y --downloadonly docker",  # have docker package in place for quick install
+                    ]
+                )
+            )
+            # Set RUN_IN_AIRBYTE_CI to tell gradle how to configure its build cache.
+            # This is consumed by settings.gradle in the repo root.
+            .with_env_variable("RUN_IN_AIRBYTE_CI", "1")
+            # Disable the Ryuk container because it needs privileged docker access which it can't have.
+            .with_env_variable("TESTCONTAINERS_RYUK_DISABLED", "true").with_workdir(self.workdir)
+            # Ensure that the .m2 directory exists.
+            .with_exec(["mkdir", "-p", self.LOCAL_MAVEN_REPOSITORY_PATH])
+        )
+
+    @staticmethod
+    def set_s3_build_cache_credentials(
+        s3_build_cache_access_key_id: Optional[dagger.Secret], s3_build_cache_secret_key: Optional[dagger.Secret]
+    ) -> Callable:
+        """Closure to set the S3 build cache credentials as environment variables on a gradle container.
+
+        Args:
+            s3_build_cache_access_key_id (Optional[dagger.Secret]): The S3 build cache access key ID as a dagger secret.
+            s3_build_cache_secret_key (Optional[dagger.Secret]): The S3 build cache secret key as a dagger secret.
+
+        Returns:
+            Callable: A function which takes a gradle container and returns a gradle container with the S3 build cache credentials set as environment variables.
+        """
```

Comment:
Again, nobody is ever going to call this, nor should we encourage them. Is this docstring really useful?